### ISSUE TITLE
Pathfinder-Luftlinie

### ISSUE THREAD

**Issue by MacRama1** [Stephan-S/FS22_AutoDrive#909]
Hallo zusammen,

habe folgendes jetzt schon öfter gehabt:

Abfahrer möchte tanken fahren und sucht den kürzesten Weg zur nächsten Tankstelle. Anscheinend richtet sich AD nach der Luftlinie denn es heißt, "konnte keinen Weg zu Tankstelle xy finden" (rot) welche auf der Luftlinie am dichtesten dran ist, er aber nicht erreichen kann weil dort keine Brücke ist. 

Würde AD die Tankstelle auf dem Streckennetz direkt suchen, käme er zur nächstmöglichen Tankstelle (schwarz). Ich habe jetzt schon einige Tankstellen deshalb auf der Map verteilt aber eine Tankestelle am Feldweg ist auch nicht sooo......
Habe den Umweg auf 10000 Meter gestellt, das hilft auch nicht.

Es wäre wirklich toll wenn dass so geändert wird das AD sich tatsächlich am Streckennetz orientiert.

![AD_Luftlinie](https://github.com/Stephan-S/FS22_AutoDrive/assets/99003248/09ffe132-1122-4d43-b1d0-0366540e580d)

Liebe Grüße
Mac

**Comment by Iwan1803** [comment]
Ja, AD sucht in Luftlinienentfernung nach der nächsten Tankstelle (oder Werkstatt), wo es im Umkreis von 15m einen Zielpunkt gibt.
Wenn kein Weg gefunden werden kann, dann scheint ja aber irgendwas am Streckennetz fehlerhaft zu sein.

**Comment by MacRama1** [comment]
Hi Iwan, habe mir sooo viele deiner how-to's und strems angesehen und mir angewöhnt nach jedem Strecken editieren den Debug zu nutzen, das Netz ist sauber...
Und ja, dasselbe mit den Werkstätten, hab davon eben so viele verteilt aber am Feldweg kommt das nicht in Frage.
Mache es jetzt halt so dass ich ein betroffenes Fahrzeug so lange fahren lasse bis es sich meldet, dann deaktiviere ich das autom. tanken/reparieren und schicke es manuell dorthin. Im Anschluss aktiviere ich die Automatik wieder damit mir das Fahrzeug nicht leer läuft.

**Comment by MacRama1** [comment]
> Ja, AD sucht in Luftlinienentfernung nach der nächsten Tankstelle (oder Werkstatt), wo es im Umkreis von 15m einen Zielpunkt gibt. Wenn kein Weg gefunden werden kann, dann scheint ja aber irgendwas am Streckennetz fehlerhaft zu sein.

Hi Iwan, habe mir sooo viele deiner how-to's und strems angesehen und mir angewöhnt nach jedem Strecken editieren den Debug zu nutzen, das Netz ist sauber...
Und ja, dasselbe mit den Werkstätten, hab davon eben so viele verteilt aber am Feldweg kommt das nicht in Frage.
Mache es jetzt halt so dass ich ein betroffenes Fahrzeug so lange fahren lasse bis es sich meldet, dann deaktiviere ich das autom. tanken/reparieren und schicke es manuell dorthin. Im Anschluss aktiviere ich die Automatik wieder damit mir das Fahrzeug nicht leer läuft.

**Comment by Iwan1803** [comment]
Kann es sein, das es zu der nördlichen Tankstelle hin sehr viele blaue Linien gibt? Möglicherweise findet AD deshalb den Weg nicht.
Du könntest natürlich auch den Zielpunkt an der "falschen" Tankstelle löschen, damit AD diese nicht anfahren kann.

**Comment by MacRama1** [comment]
> Kann es sein, das es zu der nördlichen Tankstelle hin sehr viele blaue Linien gibt? Möglicherweise findet AD deshalb den Weg nicht. Du könntest natürlich auch den Zielpunkt an der "falschen" Tankstelle löschen, damit AD diese nicht anfahren kann.

Nicht so viele blaue, es ist halt ein einspuriger Feldweg.
Das mit dem löschen würde wohl funktionieren aber dann haben die nördlichen Fahrzeuge keine Tankstelle/Werkstatt mehr.
Nun ich habe das Netz nochmal getestet, das ist sauber, abgesehen davon findet AD den Weg zur Tanke wenn ich ihn manuell hinschicke. Deshalb kann kein Fehler in Netz sein. Nebenbei, er fährt ja auch ordentlich zum Abladepunkt und der befindet sich auf derselben Strecke direkt gegenüber der Tankstelle. 
Das Phänomen tritt nur auf wenn das mit der Luftlinie passiert. Vorhin wollte es der Zufall so dass AD tanken wollte gleich nachdem er abgeladen hat, das funktionierte natürlich da er bloß die Straße überqueren brauchte.
Ich würde sagen die Luftlinie ist ne gute Idee aber funktioniert nicht wenn keine Brücke da ist.
Aber wer denkt schon an so ein Szenario wenn er ein Script schreibt.....

**Comment by Iwan1803** [comment]
AD fährt nicht einfach wild durch die Gegend sondern nutzt schon das Streckennetz, um eine Tankstelle oder Werkstatt zu erreichen 😉
Es ist auf der Map scheinbar ein wenig "problematisch", den Weg dann zu finden. Was passiert denn, wenn du ihn einfach da unten auf das Feld stellst und dann manuell zu dem Tankstellen-Wegpunkt fahren lässt? Kommt er dann problemlos an?

**Comment by MacRama1** [comment]
Nee sorry falls ich mich mich falsch ausgedrückt habe, ich wollte AD nichts schlechtes nachreden oder schlecht dastehen lassen, schließlich bin ich einer von vielen begeisterten Nutzern.

Ja, wie in meinem letzten Text beschrieben. Manuell fährt er direkt dorthin, das ist es wie ich mir helfe da ich nicht selbst fahren möchte weil die Wege ewig lang sind. 16x Map und sämtliche Brücken müssen erst errichtet werden wozu erst eine Reihe von Fabriken gebaut werden müssen. Es ist noch ein langer Weg bis die 1. Brücke steht.

Also ja, manuell funktioniert es tadellos und ein Feld weiter südlich funktioniert auch die Automatik wieder richtig, da ist dann die andere Tankstelle dichter dran.

Ich denke, wenn man bei der Luftlinie im Script die Priorität runtersetzt fährt AD direkt die nächste Tankstelle oder Werkstatt auf dem Netz an. 
Vielleicht kann man die Luftlinie als Option mit einbauen, so das der Nutzer selbst einstellen kann ob die Luftlinie genutzt werden soll oder nicht.

**Comment by MacRama1** [comment]
> AD fährt nicht einfach wild durch die Gegend sondern nutzt schon das Streckennetz, um eine Tankstelle oder Werkstatt zu erreichen 😉 Es ist auf der Map scheinbar ein wenig "problematisch", den Weg dann zu finden. Was passiert denn, wenn du ihn einfach da unten auf das Feld stellst und dann manuell zu dem Tankstellen-Wegpunkt fahren lässt? Kommt er dann problemlos an?


Nee sorry falls ich mich mich falsch ausgedrückt habe, ich wollte AD nichts schlechtes nachreden oder schlecht dastehen lassen, schließlich bin ich einer von vielen begeisterten Nutzern und das schon seit LS17.

Ja, wie in meinem letzten Text beschrieben. Manuell fährt er direkt dorthin, das ist es wie ich mir helfe da ich nicht selbst fahren möchte weil die Wege ewig lang sind. 16x Map und sämtliche Brücken müssen erst errichtet werden wozu erst eine Reihe von Fabriken gebaut werden müssen. Es ist noch ein langer Weg bis die 1. Brücke steht.

Also ja, manuell funktioniert es tadellos und ein Feld weiter südlich funktioniert auch die Automatik wieder richtig, da ist dann die andere Tankstelle dichter dran.

Ich denke, wenn man bei der Luftlinie im Script die Priorität runtersetzt fährt AD direkt die nächste Tankstelle oder Werkstatt auf dem Netz an.
Vielleicht kann man die Luftlinie als Option mit einbauen, so das der Nutzer selbst einstellen kann ob die Luftlinie genutzt werden soll oder nicht.

**Comment by coatsy35** [comment]
I end up with a similar situation on the oaks with hauling to a sell point. AD does not follow the outer route around the field and then exit, a lot of the time it wants to drive as the crow flies and head through the hedge to the next field etc... when it should drive around the field using my outer parameter path (which is two way - blue) and then exit. I mainly use the autodrive guys' routes. It's like it's too keen to go direct.
